### ISSUE TITLE
Java Build Issues

### ISSUE THREAD

**Issue by crazyash** [standard-webhooks/standard-webhooks#219]
### libraries/java/build.gradle corrections

**Compatibility Version**
java.net.http is not present in Java Version 1.8. So, the compatible versions have to be updated to at least Java 11.

```
sourceCompatibility = JavaVersion.VERSION_11
targetCompatibility = JavaVersion.VERSION_11
```

**Source Set**
Test fails as the source set path is incorrect. It has to be corrected as below:

```
sourceSets {
    main.java.srcDirs = ['lib/src/main/java']
    test.java.srcDirs = ['lib/src/test/java']
}
```

**Fat Jar (Optional)**
An option can be provided to create a fat jar.

```
jar {
    archiveBaseName = POM_ARTIFACT_ID
    duplicatesStrategy = DuplicatesStrategy.EXCLUDE // add this line
    manifest {
        attributes('Implementation-Title': POM_NAME,
                   'Implementation-Version': VERSION_NAME,
                   'Bundle-SymbolicName': POM_ARTIFACT_ID,
                   'Export-Package': 'com.standardwebhooks,com.standardwebhooks.exceptions')

        archiveVersion = VERSION_NAME
    }
    // add below lines
    from {
        configurations.runtimeClasspath.collect { it.isDirectory() ? it : zipTree(it) }
    }
}
```

**Comment by tasn** [comment]
Why do we even require java.net.http, any idea? I also feel like it's something we fixed in https://github.com/svix/svix-webhooks/tree/main/java, maybe we can draw inspiration from there?

**Comment by crazyash** [comment]
> Why do we even require java.net.http, any idea? I also feel like it's something we fixed in https://github.com/svix/svix-webhooks/tree/main/java, maybe we can draw inspiration from there?

To parse the webhook headers.. Refer: https://github.com/svix/svix-webhooks/blob/main/java/lib/src/test/java/com/svix/WebhookTest.java

Without this fix, gradle build fails.

**Comment by tasn** [comment]
But what did we do as a fix in the other repo?

**Comment by svix-jplatte** [comment]
In Svix we only support verification on Java 11+, API access on 1.8.

**Comment by tasn** [comment]
Oh right, I think we agreed there that Java 11 is probably fine? I'd be happy to accept a patch that makes 8 work too, but 11 was released in 2018 (8 years ago), so probably new enough?

@crazyash, thoughts?

**Comment by crazyash** [comment]
Java 23 is the newest version. But to create a successful build with the existing code at least Java 11 is required.

**Comment by tasn** [comment]
Yeah, so we'll need to update the min requirement to 11. Though any objections there? E.g. is the Java ecosystem still big on 1.8 and we are missing out by not following it?

**Comment by tasn** [comment]
I think this is fixed in #232!

@crazyash, please let me know if that's not the case.
